File: core/src/main/java/forpleuvoir/dhwu1a/core/event/group/BotLeaveEventKick.java
package forpleuvoir.dhwu1a.core.event.group;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.common.data.GroupData;
import forpleuvoir.dhwu1a.core.user.Group;

import javax.annotation.Nullable;

import static forpleuvoir.dhwu1a.core.common.ApiKey.GROUP;

/**
 * Bot被踢出一个群
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.group
 * <p>#class_name BotLeaveEventKick
 * <p>#create_time 2021/7/3 23:09
 */
public class BotLeaveEventKick extends GroupEvent {

    /**
     * Bot被踢出的群信息
     */
    @SerializedName(GROUP)
    public final GroupData group;

    private BotLeaveEventKick(GroupData group) {
        super(GroupEventType.BotLeaveEventKick);
        this.group = group;
    }

    /**
     * 退出群之后则不再能获取到该群对象
     *
     * @return null
     */
    @Nullable
    @Override
    public Group getGroup() {
        return null;
    }

    @Override
    public String toPlainText() {
        return String.format("%s[group:%d]", type, group.id);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/group/BotLeaveEventKick.kt
package forpleuvoir.dhwu1a.core.event.group

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.GROUP
import forpleuvoir.dhwu1a.core.common.data.GroupData
import forpleuvoir.dhwu1a.core.user.Group

/**
 * Bot被踢出一个群
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.group
 *
 * #class_name BotLeaveEventKick
 *
 * #create_time 2021/7/3 23:09
 */
class BotLeaveEventKick private constructor(group: GroupData) : GroupEvent(GroupEventType.BotLeaveEventKick) {
    override fun callback() {
        bot.leaveGroup(this, null)
    }

    /**
     * Bot被踢出的群信息
     */
    @SerializedName(GROUP)
    val group: GroupData

    /**
     * 退出群之后则不再能获取到该群对象
     *
     * @return null
     */
    override fun getGroup(): Group? {
        return null
    }

    override fun toPlainText(): String {
        return String.format("%s[group:%d]", type, group.id)
    }

    init {
        this.group = group
    }
}